### ISSUE TITLE
add javadocs

### ISSUE THREAD

**Issue by ivoysey** [jasonyih/TestRepo#2]
It's always the right move to write down what your code does and why. If you don't know it, check out the javadoc syntax and use that to write comments for the class and method and implementation as you see fit.

**Comment by jasonyih** [comment]
I tried out Javadoc, and it seems more useful for organizing and describing
my code than how I did it on my programs back at school! I read online that
I could "run" Javadoc to return some HTML with all my descriptions, but how
exactly do I "run" Javadoc after writing the comments? Do I need to
download something or is it built into Java?

**Comment by ivoysey** [comment]
`javadoc` is indeed a thing built into java! There's a commandline tool described here (https://docs.oracle.com/javase/9/javadoc/javadoc-command.htm#JSJAV-GUID-C27CE557-E5C6-4688-9FA5-9E9DE886A569) but also I suspect Eclipse has built in support. It's really the standard. Play around with it and let me know if you have trouble.

By the way, if you mention an issue number in a commit message, that commit shows up in the list of comments on the issue. It's a very handy way to keep track of why which changes got made. So you might do something like `git commit -am "adding javadocs per #2"` at the command line.

**Comment by jasonyih** [comment]
I generated Javadoc for my project using Eclipse, and it was actually
really simple with Eclipse's help, so I had no trouble there. It's really
cool how it works and I actually recognize the formatting of the html pages
generated! It seems that the webpages for the API of all the Java classes
use Javadoc, I just never knew that this was the case.

I tried committing a change and referencing the issue number in the commit
message on Github, and it worked, but how do I do that via command line on
github? I couldn't find an option for the command line on the website.
Also, should I add all of the documents generated by Javadoc in Eclipse to
the Github repo?

**Comment by ivoysey** [comment]
Yeah, it's cool how uniform the look and feel is. There are downside but that's one of the upsides.

From the commandline if you just mention the number of the issue in your commit string, github does some heuristic stuff and will pick up on it 99% of the time. Use the "#" too, so something like `commit -am "great new features for #2"`.

You should add the javadocs to the repo! Then other folks can read them. If you're feeling especially enthusiastic, you should see if there's a way for them to get automatically generated when you push to the repo so that you know what people find on the repo is always going to reflect the most recent thing you wrote (instead of if they get skewed because someone forgot to regenerate them). I'm sure there's a standard answer for this question out there if you search around; you might try using the term "continuous integration" in your search query.
